CodingStandards/ProgramEntryPoint.cs

```diff
@@ -5,33 +5,48 @@
 using System.Text;
 using System.Threading.Tasks;
 
+/// <summary>
+/// Writes a celebratory new year message.
+/// </summary>
 class Program
 {
-        //This is the entry point for the program.
-        static int Main(string[] args)
-{
-    return writeMessage(dtNow, Console.Out);
-}
-
-        static DateTime dtNow = DateTime.Now;
-
-public static int writeMessage(DateTime CurrentDate, TextWriter writer)
-{
-    var intRetVal = 0; const Int32 MESSAGE_COUNT = 5; for (int MyIndex = 0; MyIndex < MESSAGE_COUNT; MyIndex++)
-    /*********************/
-    /* greater than 2016 */
-    /*********************/
-    if (CurrentDate.Year > 2016) writer.WriteLine("Happy New Year!");
-    else
-                writer.WriteLine("Hello From " + CurrentDate.Year);
-
-
-    //--------------------return the result value----------------------------
-
-
-
-
-    return intRetVal;
+    private static DateTime Now = DateTime.Now;
+
+    // This is the entry point for the program.
```

Comment:
This comment could be removed since your comment is much more helpful. This is just a suggestion, not a coding standard violation.